libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Network/Network/Action/CSFSalesforceAction.m

```diff
@@ -118,111 +118,127 @@ - (NSDictionary *)headersForAction {
 - (id)contentFromData:(NSData*)data fromResponse:(NSHTTPURLResponse*)response error:(NSError**)error {
     NSError *responseError = nil;
     id content = [super contentFromData:data fromResponse:response error:&responseError];
-
-    if (!responseError) {
-        NSObject *msgObj = nil;
-        NSString *errorCode = nil;
-        if (content) {
-            if ([content isKindOfClass:[NSArray class]] && [(NSArray*)content count] > 0) {
-                NSArray *jsonArray = (NSArray*)content;
-                NSDictionary *errorDict = jsonArray[0];
-                if ([errorDict isKindOfClass:[NSDictionary class]] && errorDict[@"errorCode"]) {
-                    msgObj = CSFSalesforceErrorMessage(errorDict);
-                    errorCode = errorDict[@"errorCode"];
-                }
-            } else if (response.statusCode >= 400 && [content isKindOfClass:[NSDictionary class]]) {
-                NSDictionary *errorDict = (NSDictionary*)content;
-                msgObj = CSFSalesforceErrorMessage(errorDict);
-                errorCode = errorDict[@"errorCode"];
-            }
+    if (responseError != nil) {
+        if (error) *error = responseError;
+        return content;
+    }
+    
+    // TODO: We need to figure out how to handle the security token some other way, so it doesn't collide with other action types.
+    CSFNetwork *network = self.enqueuedNetwork;
+    if ([content isKindOfClass:[NSDictionary class]]) {
+        NSDictionary *jsonContent = (NSDictionary*)content;
+        NSString *securityToken = jsonContent[CSFActionSecurityTokenKey]; // retrieve the CSRF security token
+        if (securityToken) {
+            network.securityToken = securityToken;
         }
-        CSFNetwork *network = self.enqueuedNetwork;
-        if (response.statusCode >= 400) {
+    }
+    
+    return content;
+}
+
+- (NSError *)errorFromData:(NSData *)data response:(NSHTTPURLResponse *)response {
+    // Check for upstream errors.
+    NSError *superError = [super errorFromData:data response:response];
+    
+    if (response.statusCode < 400) {
+        // Nothing that we care about, from an error standpoint.
+        return superError;
```

Comment:
Probably...but the inherited class intentionally doesn't care—it's an implementation detail in `CSFAction`.  If that implementation changes to have an error in that case, this code stays the same.